spec/api/read_spec.rb

```diff
@@ -0,0 +1,25 @@
+require 'spec_helper'
+
+describe "app" do
+  describe "read" do
+
+    before(:each) do
+      set_api_key_header
+      setup_10_threads
+    end
+
+    describe "POST /api/v1/users/:user_id/read" do
+      it "marks a thread as read for the user" do
+        thread = @threads["t0"]
+        user = create_test_user(42)
+        current_time = Time.now.utc.strftime("%Y-%m-%dT%H:%M:%SZ")
+        post "/api/v1/users/#{user.external_id}/read", source_type: "thread", source_id: thread.id
+        last_response.should be_ok
+        user.reload
+        read_state = user.read_states.find_or_create_by(course_id: thread.course_id)
+        read_state.last_read_times[thread.id.to_s].utc.strftime("%Y-%m-%dT%H:%M:%SZ").should == current_time
```

Comment:
If the post sets the read_state.last_read_time, this is fragile, because it introduces a race condition where the time could increment while the post is happening.  It would be better to mock the place (in `user.mark_as_read` maybe?) where the date gets set, and set it to a known value, then test against that known value.